### ISSUE TITLE
Fresh dev server suddenly panicking

### ISSUE THREAD

**Issue by roj1512** [denoland/deno#21301]
```text
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.38.2
Args: ["/home/xxxx/.deno/bin/deno", "run", "-A", "--unstable", "--watch=static/,routes/", "dev.ts"]

thread 'main' panicked at ext/http/service.rs:255:5:
HTTP state error: Expected to be last strong reference
stack backtrace:
   0: rust_begin_unwind
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:595:5
   1: core::panicking::panic_fmt
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/panicking.rs:67:14
   2: deno_http::service::HttpRecord::recycle
   3: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   4: deno_http::http_next::op_http_set_response_body_resource::v8_fn_ptr
   5: Builtins_CallApiCallbackGeneric
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by dbushell** [comment]
I am seeing a similar error:

```
Platform: macos aarch64
Version: 1.38.2
Args: ["deno", "run", "-A", "mod.ts"]

thread 'main' panicked at ext/http/service.rs:255:5:
HTTP state error: Expected to be last strong reference
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: deno_http::service::HttpRecord::recycle
   3: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   4: deno_http::http_next::op_http_set_response_body_resource::v8_fn_ptr
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

I'm not using Fresh but I am using esbuild WASM similar to how Fresh does it. This panic started occuring not long after I implemented that — but I have no idea if it's related.

**Comment by bartlomieju** [comment]
Thanks for the report. Already fixed in https://github.com/denoland/deno/pull/21252, will be released in v1.38.3 tonight.
